webapp/src/index.tsx

```diff
@@ -22,8 +25,28 @@ export default class Plugin {
                 const state = store.getState();
                 const post = getPost(state, postID);
 
-                // Don't show up if the post has no attachments. Permissions are checked server-side.
-                return typeof post.file_ids?.length !== 'undefined' && post.file_ids?.length > 0;
+                // Check if post is editable
+                const config = getConfig(state);
+                const edit_time_limit : number = config.PostEditTimeLimit ? Number(config.PostEditTimeLimit) : -1;
+                if (!(edit_time_limit !== -1 && post.create_at + (edit_time_limit * 1000) > Date.now())) {
+                    return false;
+                }
+
+                // Check if post has attachments
+                if (!(typeof post.file_ids?.length !== 'undefined' && post.file_ids?.length > 0)) {
+                    return false;
+                }
+
+                // Check if the user has permissions to edit his own post or edit other's posts if not the author
+                const user = getCurrentUser(state);
+                let permission = 'edit_post';
+                if (post.user_id !== user.id) {
+                    permission = 'delete_others_posts';
```

Comment:
Should this be `edit_others_posts`?

nit: you may import the constants, so you could do:

```javascript
import {Permissions} from 'mattermost-redux/constants';

...
let permission = Permissions.EDIT_POST;
...
permission = Permissions.EDIT_OTHERS_POSTS;
```

Reply:
I was looking for that yesterday but couldn't find it. Thank you!